File: react-type-vite/src/services/api/response/educationalUnitResponse.ts
import type { CourseResponse } from "./courseResponse";
import type { DepartmentResponse } from "./departmentResponse";


export interface EducationalUnitResponse {
  id: number;
  name: string;
  type?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  logo?: string;
  businessLicense?: string;
  businessLicenseOriginal?: string;
  businessLicenseSigned?: string;
  businessLicenseOriginalHash?: string;
  businessLicenseSignedHash?: string;
  signatureStatus?: string;
  signatureErrorCode?: string;
  signatureErrorReason?: string;
  signatureWarning?: string;
  signatureRevocationStatus?: string;
  signatureVerifiedAt?: Date;
  certificateExpiryDate?: Date;
  description?: string;
  establishedYear?: number;
  status: string;
  subscriptionStartDate?: Date;
  subscriptionEndDate?: Date;
  createdAt?: Date;
  totalDepartments?: number;
  totalCourses?: number;
  totalTeachers?: number;
  totalStudents?: number;
  departments?: Set<DepartmentResponse>;
  courses?: Set<CourseResponse>;

  // Representative information
  representativeName?: string;
  representativeEmail?: string;
  representativePhone?: string;
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/EducationalUnitResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EducationalUnitResponse {
    private Integer id;

    private String name;

    private String type;

    private String address;

    private String phone;

    private String email;

    private String website;

    // URL on cloud
    private String logo;

    public String businessLicense;

    private String businessLicenseOriginal;

    private String businessLicenseSigned;

    private String businessLicenseOriginalHash;

    private String businessLicenseSignedHash;

    private String signatureStatus;

    private String signatureErrorCode;

    private String signatureErrorReason;

    private String signatureWarning;

    private String signatureRevocationStatus;

    private Date signatureVerifiedAt;

    private Date certificateExpiryDate;

    private String description;

    private Integer establishedYear;

    private String status;
    private Date createdAt;

    private Integer totalDepartments;

    private Integer totalCourses;

    private Integer totalTeachers;

    private Integer totalStudents;

    private Set<DepartmentResponse> departments;

    // representative info
    private String representativeName;
    private String representativeEmail;
    private String representativePhone;
}
